### PR TITLE
Respect configured primary key type in migration

### PR DESCRIPTION
**Pull request by MSchmidt**
Instead of changing the migration manually this will respect the setting from the rails project. Implementation is the same as ActiveStorage uses.

### CODE REVIEW

File: db/migrate/1_create_pay_tables.rb [hunk 1]
```diff
@@ -12,8 +15,8 @@ def change
     add_index :pay_customers, [:owner_type, :owner_id, :deleted_at, :default], name: :pay_customer_owner_index
     add_index :pay_customers, [:processor, :processor_id], unique: true
 
-    create_table :pay_merchants do |t|
-      t.belongs_to :owner, polymorphic: true, index: false
+    create_table :pay_merchants, id: :uuid do |t|
```
Comment (excid3) [inline db/migrate/1_create_pay_tables.rb:18]:
```suggestion
    create_table :pay_merchants, id: primary_key_type do |t|
```

These should use the `primary_key_type` variable, right?

**Comment by excid3** [comment]
This is great @MSchmidt! Didn't realize ActiveStorage had this and had wanted a solution for a while like this.

**Comment by MSchmidt** [comment]
Oh you are absolutely right. I adjusted this in my project and then messed the PR up. Will fix

File: db/migrate/1_create_pay_tables.rb (hunk 1 above)
Comment (MSchmidt) [reply db/migrate/1_create_pay_tables.rb]: 🙈 Fixed!

**Comment by excid3** [comment]
Perfect. Thanks again for this. Super duper handy and nice to see that Rails uses this approach internally!

**Comment by MSchmidt** [comment]
Thanks for merging so quickly. Wondering if this should be part of `ActiveRecord::Migration` then. Not very DRY if every project needs the same few lines of code.

**Comment by excid3** [comment]
Sure seems like it could be extracted as a couple methods for easy access because any Rails engine with migrations _should_ use it. Maybe you could get it in before Rails 7.1?

Tag me on the PR if you make one to Rails!

**Pull request merged**